NitroxModel/DataStructures/GameLogic/SubnauticaPlayerPreferences.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using BinaryPack.Attributes;
+
+namespace NitroxModel.DataStructures.GameLogic;
+
+[Serializable, DataContract]
+public class SubnauticaPlayerPreferences
+{
+    [DataMember(Order = 1)]
+    public Dictionary<string, PingInstancePreference> PingPreferences { get; set; } = new();
+
+    [DataMember(Order = 2)]
+    public List<int> PinPreferences { get; set; } = new();
```

Comment:
What are these?

Reply:
PingPreferences will be reworked to be fixed but they contain the user's personal preference over the pings (you can change their color and whether they're visible or not in the PDA).
PinPreferences are related to blueprints pins. This is a new feature for normal Subnautica afaik, you can pin a feature and to have its recipe shown on your screen when playing normally. And you can rearrange the list to have the pins in a precise order.

Reply:
This int is representing a TechType.

Reply:
Can we just use tech type?  I believe it is serialized as an int anyways

Reply:
The point here is to avoid having the implementation of NitroxTechType which is too heavy when serialized.
We can't use this enum from UWE except if I move this class to NitroxModel-Subnautica which wouldn't be worth the required changes.
The only "solution" would be to wait for Jannify's reimplementation of NitroxTechType but that'd be for later on.

Reply:
For now can you rename to PinnedTechTypes so it is understood what the integer is?